lib/govuk_schemas/random.rb

```diff
@@ -96,6 +104,18 @@ def string_for_regex(pattern)
           DOC
         end
       end
+
+    private
+
+      def letter
+        letters = ("a".."f").to_a
+        letters[rand(0..letters.count - 1)]
+      end
+
+      def number
+        numbers = ("0".."9").to_a
+        numbers[rand(0..numbers.count - 1)]
+      end
     end
   end
 end
```

Comment:
The code in this class looks good, might be worth making it a commit of it's own about deterministic random usage so there's good emphasis on the why as that could be quite a future headscratcher (current commit looks good but is wrapped up with the seed stuff which seems distinct).

It's probably worth writing some unit tests for it too since we expect the logic to reflect the external condition of srand seeding.

Reply:
I've taken the Content Publisher approach here, where I'm confident every change in this commit is being tested thoroughly by the unit test I've added in `random_example_spec.rb`. It seems a little unnecessary to write tests explicitly for the `hex`/`uuid` methods, especially as other public methods in this class also don't have an explicit test (e.g. base_path, anchor). Let me know if you think I should write more tests.

Reply:
Sure, it's a pretty minor point. The thing that I find a little peculiar is that we've added unit test for a class that didn't actually change for this. So it feels like we're doing an integration test rather than a unit test. I don't know this code base well enough to know if that's inconsistent for this project or not though. So happy to go with your call on what feels right.